.github/workflows/ci.yml

```diff
@@ -50,7 +50,46 @@ jobs:
     needs: build-artifacts
     secrets: inherit
 
+  snownet-tests:
+    needs: build-artifacts
+    if: ${{ github.event_name == 'pull_request' || github.event_name == 'merge_group' }}
+    name: snownet-tests-${{ matrix.name }}
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+      id-token: write
+      pull-requests: write
+    env:
+      RELAY_IMAGE: 'us-east1-docker.pkg.dev/firezone-staging/firezone/relay'
+      RELAY_TAG: ${{ github.sha }}
+      SNOWNET_IMAGE: 'us-east1-docker.pkg.dev/firezone-staging/firezone/snownet-tests'
+      SNOWNET_TAG: ${{ github.sha }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - file: docker-compose.lan.yml
+            name: lan
+          - file: docker-compose.wan-hp.yml
+            name: wan-hp
+          - file: docker-compose.wan-relay.yml
+            name: wan-relay
```

Comment:
Would be nice to tidy this up and only use a regular matrix based on the name. Not blocking this PR of course.

Reply:
Fixed